actors/professionPreviewer.js

```diff
@@ -1,8 +1,12 @@
 import { DataProvider } from "./dataProvider.js";
 
 export class ProfessionPreviewer {
-    constructor(dataProvider) {
-        this.dataProvider = dataProvider;
+    dataProvider = null;
+    profession = null;
+
+    constructor(profession) {
+        this.dataProvider = new DataProvider();
```

Comment:
Please don't do work in the constructor, there's a reason why I injected the outside providers. It's ok if we auto load them here for now, but instantiating classes in a constructor is a non-negotiable

Reply:
FIxed